ports/nrf/usb/usb.c

```diff
@@ -82,6 +81,21 @@ void usb_init(void) {
         tusb_hal_nrf_power_event(NRFX_POWER_USB_EVT_READY);
     }
 
+    // create serial number based on device unique id
+    extern uint16_t usb_desc_str_serial[1 + 16];
+
+    char nibble_to_hex[16] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};
+
+    for (int i = 0; i < 2; i++) {
+        for (int j = 0; j < 8; j++) {
+            uint8_t nibble = (NRF_FICR->DEVICEID[i] >> j * 4) & 0xf;
+
+            // Invert order since it is LE, +1 for skipping descriptor header
+            uint8_t  const idx = (15 - (i * 8 + j)) + 1;
+            usb_desc_str_serial[idx] = nibble_to_hex[nibble];
```

Comment:
I thought the descriptor string had to be UTF-16, so every other byte has to be zero. Am I wrong about that? The code in `ports/atmel-samd/supervisor/serial.c` does that:
```
    for (int i = 0; i < 4; i++) {
        for (int j = 0; j < 8; j++) {
            uint8_t nibble = (*(addresses[i]) >> j * 4) & 0xf;
            // Strings are UTF-16-LE encoded.
            serial_number[i * 16 + j * 2] = nibble_to_hex[nibble];
            serial_number[i * 16 + j * 2 + 1] = 0;
        }
    }
```

Reply:
I realized that a minute ago when I stepped away from the computer :smile: